### ISSUE TITLE
Add type invariant to postcond

### ISSUE THREAD

**Issue by n-osborne** [ocaml-gospel/ortac#186]

**Comment by n-osborne** [comment]
Gospel type invariants are talking about models. As we only build the model of the type `sut` (and call it `state`), we can only check invariants on `state`

**Comment by n-osborne** [comment]
I believe a `sut` is available both in `precond` and `postcond`. My point was that if the library has two types, both with gospel type invraiants, there is no need to collect the invariants for the non-`sut` type as we don't have a model to check them against for this type. (My first though and experiment was to check invariants for all the types in the library).

Checking for preservation of invariants, I'm currently implementing checking them in the `postcond` function.
Regarding the initial value, as it is provided by the user in the `init_sut` value we can either wrap this value in a check for invariants, or assume that it is the user responsability. I think it is not costly enough to not check it.

**Comment by jmid** [comment]
> I believe a `sut` is available both in `precond` and `postcond`. 

Nope, unfortunately not: https://ocaml-multicore.github.io/multicoretests/0.3/qcheck-stm/STM/module-type-Spec/index.html
```ocaml
val precond : cmd -> state -> bool
val postcond : cmd -> state -> res -> bool
```

...but yes it seems I misunderstood you :slightly_smiling_face:

**Comment by n-osborne** [comment]
Closed by #197
